iris_doc/api_tagger.py

```diff
@@ -257,15 +257,21 @@ def _findClassScopeStartIndex(self, lines: List[str], startIndex: int) -> int:
         while (i < len(lines)):
             if self.__syntaxMatcher.matchClassScopeStart(lines[i]):
                 return i
-
             i += 1
+        return -1
 
+    def _findExtensionScopeStartIndex(self, lines: List[str], startIndex: int) -> int:
+        i = startIndex
+        while(i < len(lines)):
+            if self.__syntaxMatcher.matchExtensionScopeStart(lines[i]):
+                return i
+            i += 1
         return -1
 
     def _findClassScopeEndIndex(self, lines: List[str], scopeStartIndex: int) -> int:
-        if not lines[scopeStartIndex].strip().endswith("{"):
-            raise IndexError(
-                "The scopeStartIndex is not correct.")
+        if not self.__syntaxMatcher.matchClassScopeStart(lines[scopeStartIndex]) and \
+            self.__syntaxMatcher.matchExtensionsScopeStart(lines[scopeStartIndex]):
```

Comment:
I'm not very sure why `matchExtensionsScopeStart ` is needed here.

Reply:
because class the _getClassTokens is called for both classes and extensions checker inside `tokenize`. The start of a class or extension class scope is a bit different, not as simple as `{` in other languages. for example it could start in any of these ways:

```objc
@interface SandwichPreferences : NSObject NS_SWIFT_NAME(Sandwich.Preferences)
@protocol AgoraRtcEngineDelegate <NSObject>
@interface AgoraRtcEngineKit : NSObject
@interface AgoraRtcEngineKit
```

and the scope ends with `@end`

whereas an extension is subtly different with the brackets in the name:

```objc
@interface AgoraRtcEngineKit(Ex)
```

with the same `@end` closure.

Reply:
Thanks for the explanation, clear now.